compute/ibm_vpc.py

```diff
@@ -515,7 +523,48 @@ def delete(
         LOG.debug(resp.get_result())
         raise NodeDeleteFailure(f"Failed to remove {node_name}")
 
-    def wait_until_vm_state_running(self, instance_id: str) -> None:
+    def shutdown(self, wait: bool = False) -> None:
+        """
+        Gracefully power off the IBM Cloud VM.
+        Args:
+            wait (bool): Wait until the VM is fully powered off.
+        """
+        try:
+            if not self.node:
+                return
+            node_id = self.node["id"]
+            LOG.info(
+                f"Initiating shutdown of IBM node: {self.node['name']} (ID: {node_id})"
+            )
+            self.service.create_instance_action(node_id, type="stop")
+            if wait:
+                try:
+                    self._wait_until_vm_state(node_id, target_state="stopped")
+                except NodeError as e:
+                    LOG.error(
+                        f"Error while waiting for IBM Cloud VM to stop with error: {e}"
+                    )
+                    raise
+        except (ResourceNotFound, NetworkOpFailure, NodeError, VolumeOpFailure):
+            raise
```

Comment:
Can you add a log statement here also, so that it will be easy to debug. In below exceptions also